src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3702,7 +3702,7 @@ TypeWithState convertCollection(BoundCollectionExpression node, TypeWithAnnotati
 
                 // Record the final state
                 NullableFlowState resultState = getResultState(node, collectionKind);
-                var resultTypeWithState = TypeWithState.Create(node.Type, resultState);
+                var resultTypeWithState = TypeWithState.Create(strippedTargetCollectionType, resultState);
```

Comment:
Which test change is relevant to this?
To observe this change, I expected to see a test where the target type is a `System.Nullable<SomeCollectionType>`, such as `b ? [1] : nullableCollection` or something like that.

Reply:
The original version was propagating the type from initial binding, instead of the nullable-reinferred type which analysis says we are actually converting to, e.g. `string~[]` in the original code and `string![]` in the updated code.

Reply:
It's possible the "strippedTargetCollectionType" is wrong in nullable value type cases, I thought that it was reasonable because collection-expr values themselves are never null. But I will have to look into it.